### ISSUE TITLE
[Feature Request] Elgato Streamdeck Control

### ISSUE THREAD

**Issue by BluesBroBo** [th-ch/youtube-music#2042]
### Preflight Checklist

- [X] I use the latest version of YouTube Music (Application).
- [X] I have searched the [issue tracker](https://github.com/th-ch/youtube-music/issues) for a feature request that matches the one I want to file, without success.

### Problem Description

A plugin to control the app via an Elgato Stream Deck.

### Proposed Solution

Control should allow to:

> Play/Pause current song
> Skip current song
> Go back to previous song
> Volume Control
> Activate/Deactivate loop
> Start a specific playlist
> Shuffle
> Perhaps activate/deactivate certain plugins?

### Alternatives Considered

Alternatively, being able to control the app via a cmd would allow to create something similar.

**Comment by ArjixWasTaken** [comment]
Blocked by https://github.com/th-ch/youtube-music/pull/1909

**Comment by ArjixWasTaken** [comment]
(blocked means that another issue/PR must be first resolved/merged before this can be dealt with)

**Comment by JellyBrick** [comment]
Duplicate of #1345

**Comment by JustMrPhoenix** [comment]
Sorry for posting on a closed issue but I want the OP and people finding this issue later to know there is a way to do most of the features without waiting for a plugin/API.
What you can do right now (with the latest release version) is to use `youtubemusic` protocol to perform actions from [here](https://github.com/th-ch/youtube-music/blob/646c0d79a3a74104593050d5b767385f176680ac/src/providers/song-controls.ts#L27). What you need to do is to get a plugin that can handle this properly. I use streamdeck plugin called [ScriptDeck](https://marketplace.elgato.com/product/scriptdeck-927e59aa-b42d-4da7-84cc-8c78f4dd7e18) to start a new process with a protocol.
For example I have it run `youtubemusic://playPause` to play/pause. You can also pass arguments like this `youtubemusic://setVolume 25`.
It does have a tiny delay but most of the stuff you asked is doable with this method. Here is an example of how i have it configured
![image](https://github.com/th-ch/youtube-music/assets/14361083/d077b9af-8c3f-4172-aae0-1dca305a93aa)

**Comment by ArjixWasTaken** [comment]
@pqt in case it did not register the protocol, you can simply call the binary of youtube music and have the uri as the first argument.

that is how uri protocols work behind the scenes

e.g.

`/path/to/youtube-music "youtubemusic://playPause"`
`"C:\Users\Admin\AppData\Local\Programs\youtube-music\YouTube Music.exe" "youtubemusic://playPause"`

**Comment by ArjixWasTaken** [comment]
which is also the reason you can't get a response from them, you can only invoke them
